### PR TITLE
feat: workflow v2 — plan checks, CLI scripts, unified statuses

### PR DESCRIPTION
**Pull request by helderberto**
## Summary

- Rename `tk:verify` → `tk:check`, remove `tk:status` (dashboard built into `tk:check`)
- Add `plan.ts` and `frontmatter.ts` parser modules
- Add `tracerkit progress <slug>` and `tracerkit archive <slug>` CLI commands
- `tk:check` skill delegates mechanical work to CLI commands (progress counting, archiving)
- Skill improvements: gray area checkpoint, Current State section, research protocol, safety valve
- Fix slug commands defaulting to homedir instead of cwd
- Unify statuses to `created` → `in_progress` → `done` everywhere
- Add `install-test` CI job for PR-based package testing
- Docs updated across the board

### Review fixes (4de11f6)

- **frontmatter**: normalize CRLF; in-place field update instead of lossy round-trip
- **plan**: normalize CRLF; case-insensitive `[X]` checkbox matching
- **cli**: remove slug by index, not value (avoids dropping duplicate args)
- **archive**: atomic with rollback on failure; PRD optional with warning
- **progress**: consistent `Total: 0/0` output when no phases
- **skill**: move checkbox writes from read-only subagent to main agent (step 3b)
- **ci**: `install-test` runs parallel with `build` (`needs: [lint, test]`)
- **docs**: cli-reference reflects optional PRD in archive

## Test plan

- [ ] `npx tracerkit init` installs 3 skills (tk:prd, tk:plan, tk:check)
- [ ] `tracerkit progress <slug>` shows per-phase checkbox counts
- [ ] `tracerkit archive <slug>` moves PRD+plan to archives, updates frontmatter
- [ ] `tracerkit archive <slug>` warns and proceeds when PRD missing
- [ ] `/tk:check` without args shows feature dashboard
- [ ] `/tk:check <slug>` runs review, uses CLI for progress/archive
- [ ] CI install-test job posts install command on PR

### CODE REVIEW

File: src/frontmatter.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+  if (!match) {
+    return `---\n${field}: ${value}\n---\n${content}`;
+  }
+
+  const existing = parseFrontmatter(content);
+  existing[field] = value;
+
+  const yaml = Object.entries(existing)
+    .map(([k, v]) => `${k}: ${v}`)
```
Comment (helderberto) [inline src/frontmatter.ts]: **Bug (data-loss risk):** `updateFrontmatter` reconstrói o YAML inteiro via `parseFrontmatter` → `Object.entries().map()`. Isso é um round-trip lossy: qualquer valor com colons, aspas ou whitespace especial é corrompido na reserialização.

Fix: ao invés de reconstruir, fazer substituição in-place da linha correspondente ao `field`. Se a key não existe, append antes do `---` de fechamento. Evita o round-trip completamente.

File: src/cli.ts [hunk 2]
```diff
@@ -29,10 +35,25 @@ const USAGE = [
...
+  return defaultDir;
+}
+
+function runSlugCommand(
+  rest: string[],
+  fn: (cwd: string, slug: string) => string[],
+): string[] {
+  const slug = rest.find((a) => !a.startsWith('-'));
+  if (!slug) return ['Error: missing <slug> argument', '', ...USAGE];
+  const target = rest.filter((a) => a !== slug);
```
[the remark this answers is not in the document]
Comment (helderberto) [reply src/cli.ts]: Fixed in 4de11f6 — now removes slug by index (`findIndex` + `filter` by index), not by value.

File: src/cli.ts [hunk 3]
```diff
@@ -29,10 +35,25 @@ const USAGE = [
...
   if (pathArg) return resolve(pathArg);
-  return homedir();
+  return defaultDir;
+}
+
+function runSlugCommand(
+  rest: string[],
+  fn: (cwd: string, slug: string) => string[],
+): string[] {
+  const slug = rest.find((a) => !a.startsWith('-'));
```
[the remark this answers is not in the document]
Comment (helderberto) [reply src/cli.ts]: Fixed in 4de11f6 — uses `findIndex` + `filter` by index. Test added for the edge case where path arg equals slug value.

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -43,6 +43,26 @@ jobs:
...
+    needs: build
+    if: github.event_name == 'pull_request'
+    permissions:
+      contents: read
+      pull-requests: write
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-node@v4
+        with:
+          node-version: 22
```
[the remark this answers is not in the document]
Comment (helderberto) [reply .github/workflows/ci.yml]: Fixed in 4de11f6 — changed `needs: build` → `needs: [lint, test]` so `install-test` runs in parallel with `build` instead of after it.
